man/man3/ConfigChipset.3
.TH "ConfigChipset" 3 "Thu Aug 23 2012" "Version 1.0" "FS-UAE Gui for Linux OS" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigChipset \- 
.PP
This class is responsible for the Chipset Configuration management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <configchipset\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigChipset\fP ()"
.br
.ti -1c
.RI "string \fBgetAccuracyConfigString\fP ()"
.br
.RI "\fIRepresents the string of Accuracy to be written in Config File\&. If Accuracy is the default value, it returns an empty string\&.
.br
 accuracy = 1\&. \fP"
.ti -1c
.RI "string \fBgetAccuracyString\fP ()"
.br
.RI "\fIRepresent the effective value of Accuracy\&. \fP"
.ti -1c
.RI "string \fBgetNTSCModeConfigString\fP ()"
.br
.RI "\fIRepresents the string of NTSCMode to be written in Config file\&. If NTSCMode is the default value, it returns an empty string\&.
.br
 ntsc_mode = 0\&. \fP"
.ti -1c
.RI "string \fBgetNTSCModeString\fP ()"
.br
.RI "\fIRepresent the effective value of NTSCMode\&. \fP"
.ti -1c
.RI "string \fBgetAmigaModelConfigString\fP ()"
.br
.RI "\fIRepresents the string of AmigaModel to be written in Config file\&. If AmigaModel is the default value, it returns an empty string\&.
.br
 amiga_model = A500\&. \fP"
.ti -1c
.RI "string \fBgetAmigaModelString\fP ()"
.br
.RI "\fIRepresent the effective value of AmigaModel\&. \fP"
.ti -1c
.RI "string \fBgetKickstartFileConfigString\fP ()"
.br
.RI "\fIRepresents the string of KickstartFile to be written in Config file\&. If KickstartFile is the default value, it returns an empty string\&.
.br
 kickstart_file = path/to/kickstart\&.rom [or path/to/kickstart\&.adf]\&. \fP"
.ti -1c
.RI "string \fBgetKickstartFileString\fP ()"
.br
.RI "\fIRepresent the effective value of KickstartFile\&. \fP"
.ti -1c
.RI "string \fBgetKickstartExtFileConfigString\fP ()"
.br
.RI "\fIRepresents the string of KickstartExtFile to be written in Config file\&. If KickstartExtFile is the default value, it returns an empty string\&.
.br
 kickstart_ext_file = path/to/extended-kickstart\&.rom\&. \fP"
.ti -1c
.RI "string \fBgetKickstartExtFileString\fP ()"
.br
.RI "\fIRepresent the effective value of KickstartExtFile\&. \fP"
.ti -1c
.RI "int \fBsetParameter\fP (string parameter, string value)"
.br
.ti -1c
.RI "void \fBsetToDefaultConfiguration\fP ()"
.br
.ti -1c
.RI "bool \fBhasParameter\fP (string parameterName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "string \fBamiga_model\fP"
.br
.RI "\fISpecify the \fBAmiga\fP model you want to emulate\&. \fP"
.ti -1c
.RI "string \fBkickstart_file\fP"
.br
.RI "\fISpecify the path to a specific kickstart file\&. \fP"
.ti -1c
.RI "string \fBkickstart_ext_file\fP"
.br
.RI "\fISpecify the path to the extended kickstart ROM (for CD32, \&.\&.\&.)\&. \fP"
.ti -1c
.RI "string \fBaccuracy\fP"
.br
.RI "\fIRepreset the accuracy of the emulation\&. \fP"
.ti -1c
.RI "string \fBntsc_mode\fP"
.br
.RI "\fIRepresent the video mode of the emulation (NTSC/PAL) \fP"
.in -1c
.SH "Detailed Description"
.PP 
This class is responsible for the Chipset Configuration management\&. 

In this class are defined all the variables for the correct management of chipset configuration's tab\&.
.br
 In this tab, you can choose the type of model you want to emulate FS-UAE (A500, A500+,\&.\&.\&.,CDTV), the paths for the kickstart and extended kickstart\&.
.br
 You can also select the emulation accuracy, the refresh rate of the video (50 Hz PAL or 60 Hz NTSC) and the various types of RAM mountable\&.
.br
 The extended kickstart path explorer is only permitted if the \fBAmiga\fP model selected si CDTV or CD32\&.
.br
 In all other cases the Cd-Rom Configuration tabs appears disabled\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "\fBConfigChipset::ConfigChipset\fP ()"
.SH "Member Function Documentation"
.PP 
.SS "string \fBConfigChipset::getAccuracyConfigString\fP ()"
.PP
Represents the string of Accuracy to be written in Config File\&. If Accuracy is the default value, it returns an empty string\&.
.br
 accuracy = 1\&. 
.SS "string \fBConfigChipset::getAccuracyString\fP ()"
.PP
Represent the effective value of Accuracy\&. 
.SS "string \fBConfigChipset::getAmigaModelConfigString\fP ()"
.PP
Represents the string of AmigaModel to be written in Config file\&. If AmigaModel is the default value, it returns an empty string\&.
.br
 amiga_model = A500\&. 
.SS "string \fBConfigChipset::getAmigaModelString\fP ()"
.PP
Represent the effective value of AmigaModel\&. 
.SS "string \fBConfigChipset::getKickstartExtFileConfigString\fP ()"
.PP
Represents the string of KickstartExtFile to be written in Config file\&. If KickstartExtFile is the default value, it returns an empty string\&.
.br
 kickstart_ext_file = path/to/extended-kickstart\&.rom\&. 
.SS "string \fBConfigChipset::getKickstartExtFileString\fP ()"
.PP
Represent the effective value of KickstartExtFile\&. 
.SS "string \fBConfigChipset::getKickstartFileConfigString\fP ()"
.PP
Represents the string of KickstartFile to be written in Config file\&. If KickstartFile is the default value, it returns an empty string\&.
.br
 kickstart_file = path/to/kickstart\&.rom [or path/to/kickstart\&.adf]\&. 
.SS "string \fBConfigChipset::getKickstartFileString\fP ()"
.PP
Represent the effective value of KickstartFile\&. 
.SS "string \fBConfigChipset::getNTSCModeConfigString\fP ()"
.PP
Represents the string of NTSCMode to be written in Config file\&. If NTSCMode is the default value, it returns an empty string\&.
.br
 ntsc_mode = 0\&. 
.SS "string \fBConfigChipset::getNTSCModeString\fP ()"
.PP
Represent the effective value of NTSCMode\&. 
.SS "bool \fBConfigChipset::hasParameter\fP (stringparameterName)"
.SS "int \fBConfigChipset::setParameter\fP (stringparameter, stringvalue)"
.SS "void \fBConfigChipset::setToDefaultConfiguration\fP ()"
.SH "Member Data Documentation"
.PP 
.SS "string \fBConfigChipset::accuracy\fP\fC [private]\fP"
.PP
Represet the accuracy of the emulation\&. 
.SS "string \fBConfigChipset::amiga_model\fP\fC [private]\fP"
.PP
Specify the \fBAmiga\fP model you want to emulate\&. 
.SS "string \fBConfigChipset::kickstart_ext_file\fP\fC [private]\fP"
.PP
Specify the path to the extended kickstart ROM (for CD32, \&.\&.\&.)\&. 
.SS "string \fBConfigChipset::kickstart_file\fP\fC [private]\fP"
.PP
Specify the path to a specific kickstart file\&. 
.SS "string \fBConfigChipset::ntsc_mode\fP\fC [private]\fP"
.PP
Represent the video mode of the emulation (NTSC/PAL) 

.SH "Author"
.PP 
Generated automatically by Doxygen for FS-UAE Gui for Linux OS from the source code\&.


configchipset.cpp
/*! \file configchipset.cpp
 configchipset.cpp

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 Written by Riva Alessandro Mario, alessandromario.riva@gmail.com
 and Sanvito Davide, dsanvito90@gmail.com. */
#include "configchipset.h"

const string DEFAULTACCURACY = "1";
const string DEFAULTNTSCMODE = "0";
const string DEFAULTAMIGAMODEL = "A500";
const string DEFAULTKICKSTARTFILE = "";
const string DEFAULTKICKSTARTEXTFILE = "";

ConfigChipset::ConfigChipset()
{
    setToDefaultConfiguration();
}

//NB se il valore è quello di default devo ritornare la stringa vuota almeno so che non devo salvarlo nella config che tanto è un inutile spreco
string ConfigChipset::getAccuracyConfigString()
{
    if (accuracy.compare(DEFAULTACCURACY)==0){return "";}
    else {return "accuracy = " + accuracy;}
}

string ConfigChipset::getAccuracyString()
{
    return accuracy;
}

string ConfigChipset::getNTSCModeConfigString()
{
    if (ntsc_mode.compare(DEFAULTNTSCMODE)==0){return "";}
    else {return "ntsc_mode = 1";}
}

string ConfigChipset::getNTSCModeString()
{
    return ntsc_mode;
}

string ConfigChipset::getAmigaModelConfigString()
{
    if (amiga_model.compare(DEFAULTAMIGAMODEL)==0){return "";}
    else {return "amiga_model = " + amiga_model;}
}

string ConfigChipset::getAmigaModelString()
{
    return amiga_model;
}

string ConfigChipset::getKickstartFileConfigString()
{
    if (kickstart_file.compare(DEFAULTKICKSTARTFILE)==0){return "";}
    else {return "kickstart_file = " + kickstart_file;}
}

string ConfigChipset::getKickstartFileString()
{
    return kickstart_file;
}

string ConfigChipset::getKickstartExtFileConfigString()
{
    if (kickstart_ext_file.compare(DEFAULTKICKSTARTEXTFILE)==0){return "";}
    else {return "kickstart_ext_file = " + kickstart_ext_file;}
}

string ConfigChipset::getKickstartExtFileString()
{
    return kickstart_ext_file;
}

int ConfigChipset::setParameter(string parameter, string value)
{
    if(parameter.compare("accuracy")==0){
        if ((value.compare("-1")==0)||(value.compare("0")==0)||(value.compare("1")==0)){
            accuracy=value;}
        else{
            accuracy=DEFAULTACCURACY;
            return -1;
        }
    } else if(parameter.compare("ntsc_mode")==0){
        if ((value.compare("0")==0)||(value.compare("1")==0)){
            ntsc_mode=value;
        }else{
            ntsc_mode=DEFAULTNTSCMODE;
            return -1;
        }
    } else if(parameter.compare("amiga_model")==0){
        if ((value.compare("A500")==0)||(value.compare("A500+")==0)||(value.compare("A600")==0)||(value.compare("A1200")==0)||(value.compare("A1200/020")==0)||(value.compare("A4000/040")==0)||(value.compare("CD32")==0)||(value.compare("CDTV")==0)){
            amiga_model=value;}
        else{
            amiga_model=DEFAULTAMIGAMODEL;
            return -1;
        }
    } else if(parameter.compare("kickstart_file")==0){
        kickstart_file=value;
    } else if(parameter.compare("kickstart_ext_file")==0){
        kickstart_ext_file=value;
    }
    return 0;
}

void ConfigChipset::setToDefaultConfiguration()
{
    amiga_model=DEFAULTAMIGAMODEL;
    accuracy=DEFAULTACCURACY;
    kickstart_file=DEFAULTKICKSTARTFILE;
    kickstart_ext_file=DEFAULTKICKSTARTEXTFILE;
    ntsc_mode=DEFAULTNTSCMODE;
}

bool ConfigChipset::hasParameter(string parameterName)
{
    return (parameterName.compare("amiga_model")==0)||(parameterName.compare("accuracy")==0)||(parameterName.compare("kickstart_file")==0)||(parameterName.compare("kickstart_ext_file")==0)||(parameterName.compare("ntsc_mode")==0);
}
